ms2deepscore/models/load_model.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Tuple, Union
+import h5py
+import json
+from tensorflow.python.keras.saving import hdf5_format
+
+from ms2deepscore.SpectrumBinner import SpectrumBinner
+from .SiameseModel import SiameseModel
+
+
+def load_model(filename: Union[str, Path]) -> SiameseModel:
+    """
+    Load a MS2DeepScore model (SiameseModel) from file.
+
+    For example:
+
+    .. code-block:: python
+
+        from ms2deepscore.models import load_model
+        model = load_model("model_file_xyz.hdf5")
+
+    Parameters
+    ----------
+    filename
+        Filename. Expecting saved SiameseModel.
+
+    """
+    with h5py.File(filename, mode='r') as f:
+        binner_json = f.attrs['spectrum_binner']
+        keras_model = hdf5_format.load_model_from_hdf5(f)
+
+    # Reconstitute spectrum_binner
+    binner_dict = json.loads(binner_json)
+    spectrum_binner = SpectrumBinner(binner_dict["number_of_bins"],
+                                     binner_dict["mz_max"], binner_dict["mz_min"],
+                                     binner_dict["peak_scaling"],
+                                     binner_dict["allowed_missing_percentage"])
+    spectrum_binner.peak_to_position = {int(key): value for key, value in binner_dict["peak_to_position"].items()}
+    spectrum_binner.known_bins = binner_dict["known_bins"]
```

Comment:
Maybe this deserialization code should live in the same module as `SpectrumBinner`? (In case you want to reuse it somehwere) Same goes for the serialisation code in `SiameseModel.save()`. You could for example have a `from_json()` class method and `to_json()` method for `SpectrumBinner` class?

Reply:
Good ideas!
I added a `to_json()` method.
Not sure about the `from_json()` method though. How would you implement that? I mean, how would you deal with the __init__ part then?

Reply:
As an alternative I could move the part you marked here into its own helper function in `load_model.py`.

Reply:
For now I left it in `load_model` but organized it a bit clearer in a sub-function.

Reply:
I totally like what you did now, so you can leave it like that (although I would vote for the `binner_from_json` function to move to the spectrumbinner module so the deserialisation code lives close to the class).

Just to show you how you could do this using a classmethod:
```python
class SpectrumBinner:
    def __init__():
         # The normal constructor
    @classmethod
    def from_json(json_str: str):
        binner_dict = json.loads(binner_json)
        spectrum_binner = cls(binner_dict["number_of_bins"],
                                         binner_dict["mz_max"], binner_dict["mz_min"],
                                         binner_dict["peak_scaling"],
                                         binner_dict["allowed_missing_percentage"])
        spectrum_binner.peak_to_position = {int(key): value for key, value in binner_dict["peak_to_position"].items()}
        spectrum_binner.known_bins = binner_dict["known_bins"]
        return spectrum_binner
        
spectrum_binner = SpectrumBinner(*args, **kwargs)  # Normal construction
spectrum_binner = SpectrumBinner.from_json(json_str) # Construction from json
```

Reply:
That's indeed nicer! So I switched to that classmethod.